lib/gitlab/client/environments.rb

```diff
@@ -0,0 +1,88 @@
+class Gitlab::Client
+  # Defines methods related to environments.
+  # @see https://docs.gitlab.com/ce/api/environments.html
+  module Environments
+    # Gets a list of project environments.
+    #
+    # @example
+    #   Gitlab.environments(5)
+    #   Gitlab.environments(5, { per_page: 10, page:  2 })
+    #
+    # @param  [Integer, String] project The ID or name of a project.
+    # @param  [Hash] options A customizable set of options.
+    # @option options [Integer] :page The page number.
+    # @option options [Integer] :per_page The number of results per page.
+    # @return [Array<Gitlab::ObjectifiedHash>]
+    def environments(project, options={})
+      get("/projects/#{url_encode project}/environments", query: options)
+    end
+
+    # Gets a single environment.
+    #
+    # @example
+    #   Gitlab.environment(5, 36)
+    #
+    # @param  [Integer, String] project The ID or name of a project.
+    # @param  [Integer] id The ID of an environment.
+    # @return [Gitlab::ObjectifiedHash]
+    def environment(project, id)
+      get("/projects/#{url_encode project}/environments/#{id}")
+    end
+
+    # Create an environment.
+    #
+    # @examples
+    #   Gitlab.create_pipeline(5, 'test-branch')
+    #   Gitlab.create_pipeline(5, 'test-branch', 'https://test-branch.example.host.com')
+    #
+    # @param  [Integer, String] project The ID or name of a project.
+    # @param  [String] env_name Name for the environment
+    # @param  [String] external_url Optional URL for viewing the deployed project in this environment
+    # @return [Gitlab::ObjectifiedHash] The updated environment.
+    def create_environment(project, env_name, external_url=nil)
```

Comment:
I would recommend we use an `options` (default empty) hash as the 3rd argument.  Since `external_url` is optional and more optional parameters _could_ be added in the future this should reduce the need to change the method signature.